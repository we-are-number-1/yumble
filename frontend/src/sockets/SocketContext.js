import React, {useState} from 'react';
import {io} from 'socket.io-client';

const SocketContext = React.createContext();

/**
 * @param {*} param0
 * @return {*} SocketContext Provider
 */
function SocketContextProvider({children}) {
  const [socket, setSocket] = useState(() => {
    setSocket(io());
  });
  const [countdown, setCountdown] = useState(null);
  const [users, setUsers] = useState(null);
  const [preferences, setPreferences] = useState(null);

  const context = {
    socket,
    countdown,
    setCountdown,
    users,
    setUsers,
    preferences,
    setPreferences,
  };

  return (
    <SocketContext.Provider value={context}>
      {children}
    </SocketContext.Provider>
  );
}

export {
  SocketContext,
  SocketContextProvider,
};
