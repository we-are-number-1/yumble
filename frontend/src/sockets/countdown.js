/**
 * Function for processing game start countdown
 * @param {*} socket
 * @param {*} cb call back function
 */
export function countdown(socket, cb) {
  socket.on('countdown', ({count}) => {
    // Call back function with current count
    cb(count);
  });
}
