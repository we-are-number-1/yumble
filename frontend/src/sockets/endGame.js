/**
 * Function for when the game ends.
 * @param {*} socket
 * @param {*} cb call back function
 */
export function endGame(socket, cb) {
  socket.on('end_game', (data) => {
    socket.disconnect(true);
    cb(data);
  });
}
