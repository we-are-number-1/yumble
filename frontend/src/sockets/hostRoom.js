/**
 * Emits a host_room even to backend sockets
 * @param {*} socket
 * @param {*} sessionId
 */
export function hostRoom(socket, sessionId) {
  socket.emit('host_room', {sessionId});
}
