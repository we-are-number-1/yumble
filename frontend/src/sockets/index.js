export {getSocketClient} from './socketClient';
export {newUser} from './newUser';
