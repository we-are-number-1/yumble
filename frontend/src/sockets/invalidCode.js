/**
 * Function for when invalid game code is entered
 * @param {*} socket
 * @param {*} cb call back function
 */
export function invalidCode(socket, cb) {
  socket.on('invalid_code', (data) => {
    cb(data);
  });
}
