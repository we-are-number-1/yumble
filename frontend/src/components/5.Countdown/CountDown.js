import { Redirect } from 'react-router-dom';
import React, { useState, useEffect, useContext } from 'react';
import Help from '../Common/Help';
import { SocketContext } from './../../sockets/SocketContext';
import * as SocketEvents from './../../sockets';
import '../Common/Help.css';
import Button from 'react-bootstrap/Button';

/**
 * @param  {*} props
 * @return {*}
 *
 * Intermission screen to countdown before the game begins.
 * Counts down from 5.
 */
const CountDown = (props) => {
  const socketContext = useContext(SocketContext);
  const [ButtonPopup, setButtonPopup] = useState(false);
  const [seconds, setSeconds] = useState(socketContext.countdown);
  const [data] = useState(props.location.state);
  const [redirect, setRedirect] = useState(false);

  /**
   * @param  {*} data
   *
   */
  function cb(data) {
    setRedirect(true);
    socketContext.setCountdown(data.nextRoundTime / 1000);
  }

  useEffect(() => {
    document.title = 'Go!';
    SocketEvents.nextRound(socketContext.socket, cb);
  }, []);

  useEffect(() => {
    let t = null;
    if (seconds >= 0) {
      t = setTimeout(() => setSeconds(seconds - 1), 1000);
    }

    return () => clearTimeout(t);
  }, [seconds]);

  return (
    <>
      <h1 className='Title'>yumble</h1>
      <div className='MakeCentre'>
        <div>
          {seconds == 0 ? (
            <div className='StartTitle'>Go!</div>
          ) : (
            <div className='StartTitle'>{seconds}</div>
          )}
        </div>
      </div>
      <Button
        onClick={() => setButtonPopup(true)}
        variant='info'
        size='lg'
        id='HelpButton'
      >
        Help
      </Button>
      <Help trigger={ButtonPopup} setTrigger={setButtonPopup}>
        <p className='MakeTextCentre'>Please wait for the timer!</p>
      </Help>
      {redirect ? (
        <Redirect to={{ pathname: '/Swiping', state: [data] }} />
      ) : null}
      <div id='dummyMap' style={{ visibility: 'hidden' }}></div>
    </>
  );
};

export default CountDown;
