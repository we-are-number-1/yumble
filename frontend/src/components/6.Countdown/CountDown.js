import {Redirect} from 'react-router-dom';
import React, {useState, useEffect, useContext} from 'react';
import Help from '../Common/Help';
import {SocketContext} from './../../sockets/SocketContext';
import * as SocketEvents from './../../sockets';
import '../Common/Help.css';

const CountDown = (props) => {
  // const history = useHistory();
  const socketContext = useContext(SocketContext);
  const [ButtonPopup, setButtonPopup] = useState(false);
  const [seconds, setSeconds] = useState(socketContext.countdown);
  const [data] = useState(props.location.state);
  const [redirect, setRedirect] = useState(false);

  /**
   *
   * @param {*} data
   */
  function cb(data) {
    setRedirect(true);
    socketContext.setCountdown(data.nextRoundTime);
  };

  useEffect(() => {
    document.title = 'Go!';
    SocketEvents.nextRound(socketContext.socket, cb);
    console.log(socketContext.countdown);
    console.log(props.location.state);
  }, []);

  useEffect(() => {
    seconds >= 0 ? setTimeout(() => setSeconds(seconds - 1), 1000) : null;
  }, [seconds]);


  return (
    <>
      <h1 className='Title'>yumble</h1>
      <div className='MakeCentre'>
        <div>
          {seconds == 0 ? (
            <div className='StartTitle'>Go!</div>
          ) : (
            <div className='StartTitle'>{seconds}</div>
          )}
        </div>
      </div>
      <button
        onClick={() => setButtonPopup(true)}
        className='SmallBtn'
        id='HelpButton'
      >
        help?
      </button>
      <Help trigger={ButtonPopup} setTrigger={setButtonPopup}>
        <p className='MakeTextCentre'>
          Please wait for the timer!
        </p>
      </Help>
      {redirect ?
        <Redirect to={{pathname: '/Swiping',
          state: [data]}}
        /> : null}
      <div id="dummyMap" style={{visibility: 'hidden'}}></div>
    </>
  );
};

export default CountDown;
