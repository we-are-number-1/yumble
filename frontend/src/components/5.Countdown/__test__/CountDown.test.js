import React from 'react';
import renderer from 'react-test-renderer';
import CountDown from '../CountDown';
import { SocketContextProvider } from '../../../sockets/SocketContext';

jest.mock('./../../../sockets/SocketContext');
jest.mock('./../../../sockets');

test('CountDown snapshot testing', () => {
  const tree = renderer
    .create(
      <SocketContextProvider>
        <CountDown location={location} />
      </SocketContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
