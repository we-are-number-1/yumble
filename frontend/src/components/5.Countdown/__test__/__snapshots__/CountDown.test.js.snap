// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDown snapshot testing 1`] = `
Array [
  <h1
    className="Title"
  >
    yumble
  </h1>,
  <div
    className="MakeCentre"
  >
    <div>
      <div
        className="StartTitle"
      />
    </div>
  </div>,
  <button
    className="btn btn-info btn-lg"
    disabled={false}
    id="HelpButton"
    onClick={[Function]}
    type="button"
  >
    Help
  </button>,
  "",
  <div
    id="dummyMap"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />,
]
`;
